tests/integration/security/https_jwt/https_jwt_test.go

```diff
@@ -0,0 +1,148 @@
+)
+
+const (
+	authHeaderKey = "Authorization"
+)
+
+// TestJWTHTTPS tests the requestauth policy with https jwks server.
+func TestJWTHTTPS(t *testing.T) {
+	payload1 := strings.Split(jwt.TokenIssuer1, ".")[1]
+
+	framework.NewTest(t).
+		Features("security.authentication.jwt").
+		Run(func(t framework.TestContext) {
+			ns := apps.Namespace1
+			istioSystemNS := istio.ClaimSystemNamespaceOrFail(t, t)
+			args := map[string]string{"Namespace": istioSystemNS.Name()}
+			applyYAML := func(filename string, ns namespace.Instance) []string {
+				policy := tmpl.EvaluateAllOrFail(t, args, file.AsStringOrFail(t, filename))
+				for _, cluster := range t.AllClusters() {
+					t.ConfigKube(cluster).ApplyYAMLOrFail(t, ns.Name(), policy...)
+				}
+				return policy
+			}
+
+			jwtServer := applyYAML("../../../../samples/jwt-server/jwt-server.yaml", istioSystemNS)
+			for _, cluster := range t.AllClusters() {
+				defer t.ConfigKube(cluster).DeleteYAMLOrFail(t, istioSystemNS.Name(), jwtServer...)
+			}
+
+			for _, cluster := range t.AllClusters() {
+				fetchFn := kube.NewSinglePodFetch(cluster, istioSystemNS.Name(), "app=jwt-server")
+				_, err := kube.WaitUntilPodsAreReady(fetchFn)
+				if err != nil {
+					t.Fatalf("pod is not getting ready : %v", err)
+				}
+			}
+
+			for _, cluster := range t.AllClusters() {
+				if _, _, err := kube.WaitUntilServiceEndpointsAreReady(cluster, istioSystemNS.Name(), "jwt-server"); err != nil {
+					t.Fatalf("Wait for jwt-server server failed: %v", err)
+				}
+			}
+
+			callCount := 1
+			if t.Clusters().IsMulticluster() {
+				// so we can validate all clusters are hit
+				callCount = util.CallsPerCluster * len(t.Clusters())
+			}
+
+			t.NewSubTest("jwt-authn").Run(func(t framework.TestContext) {
+				testCase := authn.TestCase{
+					Name:   "valid-token-forward-remote-jwks",
+					Config: "remotehttps",
+					CallOpts: echo.CallOptions{
+						PortName: "http",
+						Scheme:   scheme.HTTP,
+						Headers: map[string][]string{
+							authHeaderKey: {"Bearer " + jwt.TokenIssuer1},
+						},
+						Path:  "/valid-token-forward-remote-jwks",
+						Count: callCount,
+					},
+					ExpectResponseCode: response.StatusCodeOK,
+					ExpectHeaders: map[string]string{
+						authHeaderKey:    "Bearer " + jwt.TokenIssuer1,
+						"X-Test-Payload": payload1,
+					},
+					// This test does not generate cross-cluster traffic, but is flaky
+					// in multicluster test. Skip in multicluster mesh.
+					// TODO(JimmyCYJ): enable the test in multicluster mesh.
+					SkipMultiCluster: true,
+				}
+
+				if testCase.SkipMultiCluster && t.Clusters().IsMulticluster() {
+					t.Skip()
+				}
+				echotest.New(t, apps.All).
+					SetupForDestination(func(t framework.TestContext, dst echo.Instances) error {
+						if testCase.Config != "" {
+							policy := yml.MustApplyNamespace(t, tmpl.MustEvaluate(
+								file.AsStringOrFail(t, fmt.Sprintf("../testdata/requestauthn/%s.yaml.tmpl", testCase.Config)),
```

Comment:
please just create a new testdata directory together with the test and put the yaml file there, you're already creating a new directory (`http_jwt`) so there is no need for using the old one.

Reply:
so is it okay if I just move **remotehttps.yaml.tmpl** file to **https_jwt** directory ?

Reply:
yes, I meant you should only move remotehttps.yaml.tmpl to https_jwt/testdata.

Reply:
sure, understood.